persistence/repository.go

```diff
@@ -80,23 +81,84 @@ func (h DynamoRepository) GetByID(id string) (*Resource, error) {
 	return resource, nil
 }
 
-// UpdateItem - Replaces an existing resource in the repository
-func (h DynamoRepository) UpdateItem(resource *Resource) error {
-	row, err := dynamodbattribute.MarshalMap(resource)
+// UpdateItem - Compares provided resource to existing resource and calls the appropriate update method
+func (h DynamoRepository) UpdateItem(resource *Resource, originalResource *Resource) error {
+
+	var err error
+
+	res := reflect.ValueOf(resource).Elem()
+	typeOfResource := res.Type()
+	orig := reflect.ValueOf(originalResource).Elem()
+
+	for i := 0; i < res.NumField(); i++ {
+		resourceField := res.Field(i)
+		origResourceField := orig.Field(i)
+
+		fieldName := typeOfResource.Field(i).Name
+
+		if resourceField.Interface() != origResourceField.Interface() {
+			switch resourceField.Type().String() {
+			case "bool":
+				err = h.updateBool(resource.ID, fieldName, resourceField.Interface().(bool))
+			default:
+				err = h.updateString(resource.ID, fieldName, resourceField.Interface().(string))
```

Comment:
It looks to me like this is creating N updates to DynamoDB where N is the number of changed fields.  Would it be possible to make an `UpdateItemInput` set all the values at once?

Reply:
Yes, I actually noted that specifically in standup a few days ago based on ensuring a "clean" update statement. This is an area for refactoring soon for sure. The issue I see (for now) with doing this at once is the complexity of the different types (string, bool, etc) and the unknown number of place holders, so it would have to address the ':r' part of this in a dynamic way. 

Not that it isn't the goal, but wanted to get this first part addressed initially.